### PR TITLE
Add toolkit support for theme.json file splitting and theme.json compilation

### PR DESCRIPTION
**Pull request by christianc1**
#### Proposal

As we embrace theme.json and full site editing, the potential for theme.json files to become too large and unwieldy is a concern. This PR adds support for splitting theme.json into multiple files for organization.

Further, the ability to compile a `theme.json` file from source at build time allows us to do things like create a global settings configuration and multiple different style configurations, then select a single style configuration at build time.

What is added:

1. Support for a new "`theme.json` Parts" setting (theme package.json)
````json
{
  "10up-toolkit": {
    "themeJsonParts": {
      "default": [
        "./theme-config/base.json",
        "./theme-config/settings.json",
        "./theme-config/styles.json"
      ],
      "no-styles": [
        "./theme-config/base.json",
        "./theme-config/settings.json"
      ]
  }
}
````

2. Support for a command line argument to allow user to define a build time which JSON files to use
````
$ npm run build -- --theme-json=default
$ npm run build -- --theme-json=no-styles
$ 10up-toolkit build --theme-json=default
````

3. A webpack plugin that, if the above values are set, will deep merge (`Object.assign`) the json object and emit a `theme.json` to the theme root.

### Possible Drawbacks

Overwrites `theme.json`.  This process is deliberate though, so it should be fine.

### Verification Process

Tested locally, if we agree this is something we want in toolkit, I'll see if I can add some tests, docs, and create a changeset.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests passed.
- [ ] I have added a changeset to my PR. See [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document for instructions

### CODE REVIEW

**Comment by christianc1** [comment]
In [0bf4b0b](https://github.com/10up/10up-toolkit/pull/222/commits/0bf4b0b7a226fbc1e7a2791b283a807cddad54cb), per request from @fabiankaegy and @joesnellpdx, support for jsonc (comments in JSON files) was added to this PR.

**Comment by joesnellpdx** [comment]
@christianc1 

Two considerations:

1) guidance added in readme (unless maybe this should be added to wp-scaffold once this is updated in toolkit)
2) In readme include your example .. but also include and example where we may have two separate "themes" with different styles ... something like the following (which is more what we'd do on EG 360):

```
"10up-toolkit": {
    "themeJsonParts": {
      "default": [
        "./theme-config/base.json",
        "./theme-config/settings.json",
        "./theme-config/styles.json"
      ],
      "brandsecondary": [
        "./theme-config/brandsecondary/base.json",
        "./theme-config/settings.json",
        "./theme-config/brandsecondary/styles.json"
      ]
  }
```

3) Think we should have an example of what these json parts would look like - or is that understood?
4) Any guidance for CI implementation for multisite vs separate websites with same repo (like EG 360)?
5) I think local guidance is self explanatory - just have to run the flag .... will it work on watch?


I don't have time to test, but wanted to be sure I added this to the review.

Otherwise, If it works... LGTM! LOVE it!

File: projects/10up-theme/theme-config/settings.jsonc [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+  // Define $schema so that autocomplete and inline validation works in supported IDEs.
+  "$schema": "https://schemas.wp.org/wp/6.0/theme.json",
```
Comment (joesnellpdx) [inline projects/10up-theme/theme-config/settings.jsonc:3]: @christianc1 

I think this is duped from base.jsonc - only need to declare schema once

Comment (christianc1) [reply projects/10up-theme/theme-config/settings.jsonc:3]: You're right, it's duplicated in both files, but IDE's that support $schema validation use that attribute to provide inline autocomplete, validation, and typehinting.  So it's defined in all the files so you can take advantage of that functionality regardless of what file you're editing.  The compiled version only includes it once.

**Comment by nicholasio** [comment]
@christianc1 Thanks for your work on this.

@fabiankaegy @joesnellpdx Is this something we should still consider adding to toolkit?

**Pull request closed without merging**